### ISSUE TITLE
Generated python code for array fields does not work

### ISSUE THREAD

**Issue by jgosmann** [mavlink/mavlink#75]
The generated Python code will fail on decoding array fields like `<field type="double[16]" name="data">...</field>` with the following stack trace

```
Traceback (most recent call last):
  File "mavlinkMonitor.py", line 14, in <module>
    msgs = mav.parse_buffer(ser.read(64))
  File "/Volumes/Home/blubb/Documents/programming/uni/softlanding/ardusensors/mavlink.py", line 2725, in parse_buffer
    m = self.parse_char(s)
  File "/Volumes/Home/blubb/Documents/programming/uni/softlanding/ardusensors/mavlink.py", line 2716, in parse_char
    m = self.decode(mbuf)
  File "/Volumes/Home/blubb/Documents/programming/uni/softlanding/ardusensors/mavlink.py", line 2777, in decode
    tlist[i] = t[order_map[i]]
IndexError: list index out of range
```

The problem is that the [unpack function in this line](https://github.com/mavlink/mavlink/blob/master/pymavlink/generator/mavgen_python.py#L356) will unpack each value of the array as a single value and not as a list or tuple of values.

Maybe I'll be able to fix it myself, but right now I'm not seeing an easy solution (and I'm new to mavlink).

**Comment by flixr** [comment]
This was fixed with #140 ?

**Comment by amilcarlucas** [comment]
No answer for over three years, I think this can be closed now.
